File: src/lib/github.ts
import { apiFetch } from "@/lib/api";

export interface RepoInfo {
  owner: string;
  repo: string;
  descricao: string;
  defaultBranch: string;
  linguagens: string[];
  stars: number;
  visibilidade: string;
  url: string;
  atualizadoEm: string | null;
  issuesAbertas: number;
}

export function obterRepoInfo(owner: string, repo: string): Promise<RepoInfo> {
  return apiFetch<RepoInfo>(
    `/api/github/repo?owner=${encodeURIComponent(owner)}&repo=${encodeURIComponent(repo)}`,
  );
}

File: backend/app/schemas/github.py
from app.schemas.common import CamelOut


class RepoInfoOut(CamelOut):
    owner: str
    repo: str
    descricao: str
    default_branch: str
    linguagens: list[str] = []
    stars: int
    visibilidade: str
    url: str
    atualizado_em: str | None = None
    issues_abertas: int
